app/scripts/metamask-controller.js

```diff
@@ -1461,12 +1461,19 @@ export default class MetamaskController extends EventEmitter {
         },
         showInAppNotification: {
           method: (origin, args) => {
-            const { message } = args;
+            const { message, title, footerLink, interfaceId } = args;
+
+            const detailedView = {
+              title,
+              ...(footerLink ? { footerLink } : {}),
```

Comment:
I'd like to keep the object clean without `undefined`. The performance difference of creating the intermediate object is neglible.

Reply:
This is not a performance worry, but it adds additional complexity that makes it harder to reason about the code.

I'm not necessarily against it, but why do we need `undefined` to not be present here?

Reply:
Technically it's not type correct if `footerLink` or `detailedView` is present on the notification object, also I have to do more type-fu to access the `detailedView` property instead of using `hasProperty` as I'm currently doing in some of the components.

Reply:
I'll consider this as a nit that we can fix after if we really want. This would just improve code readability :)